src/tribol/geom/ContactPlane.cpp

```diff
@@ -2021,16 +1690,49 @@ FaceGeomError CheckEdgePair( InterfacePair& pair,
 
    // CHECK #6: compute the projected length of overlap on the contact plane.
    // At this point the edges are proximate and likely have a positive 
-   // projected length of overlap. 
+   // projected length of overlap.
+
+   if (cp.m_intermediatePlane)
+   {
+      // COMMON_PLANE normal calculation:
+      // compute the cp normal as the average of the two face normals, and in 
+      // the direction such that the dot product between the cp normal and 
+      // the normal of face 2 is positive.
+      cp.m_nX = 0.5 * (mesh2.getElementNormals()[0][ edgeId2 ] - mesh1.getElementNormals()[0][ edgeId1 ]);
+      cp.m_nY = 0.5 * (mesh2.getElementNormals()[1][ edgeId2 ] - mesh1.getElementNormals()[1][ edgeId1 ]);
+      cp.m_nZ = 0.0; // zero out the third component of the normal
+   }
+   else
+   {
+      // MORTAR normal calculation. This is the normal of the nonmortar surface
+      cp.m_nX = mesh2.getElementNormals()[0][ edgeId2 ];
+      cp.m_nY = mesh2.getElementNormals()[1][ edgeId2 ];
+      cp.m_nZ = 0.;
+   }
+
+   // normalize the cp normal
+   RealT mag;
+   RealT invMag;
 
-   cp.computeNormal(); 
```

Comment:
Verify that this can be deleted and/or moved. We may have to locate the plane point early on? Replicate what was done before?